password_enhancements.module

```diff
@@ -0,0 +1,416 @@
+<?php
+
+/**
+ * @file
+ * Password related enchantments.
+ */
+
+use Drupal\Component\Render\FormattableMarkup;
+use Drupal\Core\Entity\EntityInterface;
+use Drupal\Core\Entity\EntityTypeInterface;
+use Drupal\Core\Field\BaseFieldDefinition;
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\password_enhancements\Form\Settings;
+use Drupal\user\UserInterface;
+
+/**
+ * Implements hook_theme().
+ */
+function password_enhancements_theme($existing, $type, $theme, $path): array {
+  return [
+    'password_enhancements_error_messages' => [
+      'variables' => [
+        'error_message' => '',
+        'messages' => [],
+      ],
+    ],
+    'password_enhancements_policy_constraints' => [
+      'variables' => [
+        'constraints' => [],
+        'descriptions' => [],
+        'hidden' => FALSE,
+      ],
+    ],
+    'password_enhancements_logged_out_message' => [
+      'variables' => [
+        'message' => '',
+      ],
+    ],
+  ];
+}
+
+/**
+ * Implements hook_form_FORM_ID_alter().
+ */
+function password_enhancements_form_user_form_alter(&$form, FormStateInterface $form_state, string $form_id): void {
+  password_enhancements_form_user_register_form_alter($form, $form_state, $form_id);
+  $form['password_change_required']['#access'] = Drupal::currentUser()->hasPermission('administer user password settings');
+  array_unshift($form['actions']['submit']['#submit'], 'password_enhancements_form_user_form_alter_submit');
+}
+
+/**
+ * Submit callback for password_enhancements_form_user_form_alter().
+ *
+ * @param array $form
+ *   The form.
+ * @param \Drupal\Core\Form\FormStateInterface $form_state
+ *   The current form state.
+ */
+function password_enhancements_form_user_form_alter_submit(array &$form, FormStateInterface $form_state): void {
+  // Update last updated date for the password.
```

Comment:
the last updated

Reply:
What if I update a user's password [programmatically](https://docs.acquia.com/tutorials/fast-track-drupal-8-coding/update-users-programmatically/) - or in a different UI than the built-in forms? Should not this code run in the case as well?

Reply:
Right, let me check what can I do about that.

Reply:
Used `hook_user_presave()` to update the last password changed date.